### PR TITLE
fix: introduce suggestions for property name

### PR DESCRIPTION
**Pull request by TreffN**
This PR introduces a list with suggestions for propertyName in the editFormConfig or featureInfoFormConfig in the layer config in shogun-admin. The suggestions are determined via a WFS-DescribeFeatureType query.

Please note: the suggestion list can only be provided, when the layer is already saved!

### CODE REVIEW

File: src/Hooks/useExecuteWfsDescribeFeatureType.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    const response = await fetch(`${url}?${UrlUtil.objectToRequestString(params)}`, {
+      method: 'GET',
+      headers: layer.sourceConfig.useBearerToken ? {
+        ...defaultHeaders,
+        ...getBearerTokenHeader(client?.getKeycloak())
+      } : defaultHeaders
+    });
+
+    if (!response.ok) {
+      throw new Error('No successful response while executing a WFS-Transaction');
```
Comment (dnlkoch) [inline src/Hooks/useExecuteWfsDescribeFeatureType.ts:87]:
```suggestion
      throw new Error('No successful response while executing a WFS DescribeFeatureType');
```

File: src/App.tsx [hunk 2]
```diff
@@ -66,6 +68,24 @@ const App: React.FC = () => {
     }
   }, [setAppInfo, setUserInfo, client]);
 
+  const executeWfsDescribeFeatureType = useExecuteWfsDescribeFeatureType();
+  const getPropertyNames = useCallback(async (layerId: number | undefined) => {
```
Comment (dnlkoch) [inline src/App.tsx:72]:
```suggestion

  const getPropertyNames = useCallback(async (layerId: number | undefined) => {
```

File: src/App.tsx [hunk 3]
```diff
@@ -66,6 +68,24 @@ const App: React.FC = () => {
...
+    let response: DescribeFeatureType | undefined;
+    const propertyNames: string[] = [];
+    if (layerSuggestionList && layerId) {
+      const layer = layerSuggestionList.filter(item => item.id === layerId)[0];
```
Comment (dnlkoch) [inline src/App.tsx:76]: The state `layerSuggestionList` is actually filled within the `registerLayerIdCompletionProvider` method. Can we refactor this logic to an own method?

File: src/App.tsx [hunk 4]
```diff
@@ -66,6 +68,24 @@ const App: React.FC = () => {
...
+      if (layer) {
+        response = await executeWfsDescribeFeatureType(layer);
```
Comment (dnlkoch) [inline src/App.tsx:78]: What happens if `executeWfsDescribeFeatureType` throws an error? Can we show the error (or at least the missing option to get the list of available properties) to the user somehow?

**Review by dnlkoch — COMMENTED** [review]
Thanks @TreffN ! Thats really cool!

File: src/App.tsx (hunk 2 above)
Comment (TreffN) [reply src/App.tsx:92]: resolved in new commit

File: src/App.tsx (hunk 3 above)
Comment (TreffN) [reply src/App.tsx:96]: resolved in new commit

File: src/App.tsx (hunk 4 above)
Comment (TreffN) [reply src/App.tsx]: resolved in new commit

**Review by dnlkoch — APPROVED** [review]

**Pull request merged**